modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/ElasticSearchChecker.java

```diff
@@ -67,8 +60,10 @@ public boolean accepts(ConfigurationGenerator cg) {
     }
 
     @Override
-    public void check(ConfigurationGenerator cg) throws ConfigurationException {
-        ElasticSearchClientConfig config = getConfig(cg);
+    public void check(ConfigurationHolder configHolder) throws ConfigurationException {
+        ElasticSearchClientConfig config = getDescriptor(configHolder, CONFIG_NAME, ElasticSearchClientConfig.class,
+                // avoid XMap to fail when trying to load class value by removing class attribute
```

Comment:
I'm interested to know more about this problem linked to xmap and classes introspection (for https://jira.nuxeo.com/browse/NXP-30092), although i realized you simply moved that comment.
Checking the class can be loaded does not make sense in this context maybe?

Reply:
If I remember correctly we don't have everything in the class loader to load what the class attribute represents. Maybe @bdelbosc has more details?